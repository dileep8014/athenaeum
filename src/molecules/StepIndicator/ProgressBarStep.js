import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Text from 'atoms/Text';
import classnames from 'classnames';
import styles from './step_indicator.module.scss';

export default class ProgressBarStep extends Component {
  titleProps() {
    const {
      step: { currentStepActive }
    } = this.props;

    return {
      color: currentStepActive ? 'primary-1' : 'neutral-3',
      bold: true
    };
  }

  subtitleProps() {
    const {
      step: { currentStepActive }
    } = this.props;

    return {
      color: currentStepActive ? 'primary-1' : 'neutral-3',
      bold: currentStepActive
    };
  }

  circleClasses() {
    const {
      step
    } = this.props;

    return [
      styles['circle'],
      step.currentStepActive && styles['circle-active'],
      step.clickable && !step.currentStepActive && styles['circle-completed'],
    ];
  }

  circleWrapperClasses() {
    const {
      step
    } = this.props;

    return [
      styles['circle-wrapper'],
      step.nextStepAccessible && styles['circle-wrapper-accessible'],
    ];
  }

  textClasses() {
    const {
      staticBar,
      step
    } = this.props;

    return [
      styles['step-wrapper'],
      !staticBar && step.clickable && styles['step-wrapper-clickable'],
      !step.currentStepActive && styles['step-wrapper-inactive']
    ];
  }

  clickProps() {
    const {
      staticBar,
      navigateToPath,
      step,
    } = this.props;

    return !staticBar ? {
      onClick: () => {
        if (step.clickable) {
          navigateToPath(step.route);
        }
      }
    } : null;
  }

  render() {
    const { step } = this.props;

    return (
      <div className={styles['breadcrumb']} {...this.clickProps()}>
        <div className={classnames(this.textClasses())}>
          <Text tag='span' size={11} font='a' spaced className={styles['step-title']} {...this.titleProps()}>{ step.text }</Text>
        </div>
        <div className={classnames(...this.circleWrapperClasses())}>
          <div className={classnames(...this.circleClasses())} />
        </div>
        {
          step.subtitle && (
            <div className={styles['step-subtitle']}>
              <Text tag='span' size={9} font='b' {...this.subtitleProps()}>{step.subtitle}</Text>
            </div>
          )
        }
      </div>
    );
  }
}

ProgressBarStep.propTypes = {
  staticBar: PropTypes.bool,
  navigateToPath: PropTypes.func,
  step: PropTypes.object,
};
