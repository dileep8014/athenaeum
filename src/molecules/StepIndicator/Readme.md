### StepIndicator Example:

```jsx
    <div style={{ width: '100%' }}>
      <StepIndicator
        steps={[
          {
            text: 'Basics',
            path: '/basics',
            currentStepActive: false,
            clickable: true,
            nextStepAccessible: true
          },
          {
            text: 'Health',
            path: '/health_profile',
            currentStepActive: false,
            clickable: true,
            nextStepAccessible: true
          },
          {
            text: 'Lifestyle',
            path: '/lifestyle',
            currentStepActive: true,
            clickable: true,
            nextStepAccessible: false
          },
          {
            text: 'Pick a policy',
            path: 'comparison',
            currentStepActive: false,
            clickable: false,
            nextStepAccessible: false
          }
        ]}
        navigateToPath={() => alert('navigating to path...')}
      />
    </div>
```
