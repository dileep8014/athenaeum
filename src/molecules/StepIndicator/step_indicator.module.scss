.step-indicator {
  display: flex;
}

.breadcrumb {
  flex: 1;

  &:last-child {
    .circle-wrapper:after {
      display: none;
    }
  }
}

.step-wrapper {
  text-align: center;
  width: 100%;
  margin-bottom: ru(1);

  &-clickable {
    cursor: pointer;
  }
}

.circle-wrapper {
  display: flex;
  width: 100%;
  position: relative;
  margin-left: 48%;

  &:after {
    content: '';
    background-color: color('secondary-2');
    height: .125rem;
    pointer-events: none;
    position: absolute;
    top: 5px;
    transition: width 600ms ease-out;
    width: 0%;
  }

  &-inaccessible {
    &:after {
      background-color: color('neutral-4');
      width: 100%;
    }
  }

  &-accessible {
    &:after {
      width: 100%;
    }
  }
}

.circle {
  background: color('primary-2');
  border: 1px solid color('neutral-4');
  border-radius: 50%;
  height: ru(.5);
  margin-right: auto;
  width: ru(.5);
  pointer-events: none;
  z-index: 1;
  position: absolute;

  &-active {
    background: color('primary-2');
    border: 2px solid color('secondary-2');
  }

  &-completed {
    background: color('secondary-2');
    border: 1px solid color('secondary-2');
  }
}

@media #{$mobile-only} {
  .step-wrapper {
    &-inactive {
     opacity: 0;
    }
  }
}
