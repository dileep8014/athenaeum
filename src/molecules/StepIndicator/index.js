import React from 'react';
import PropTypes from 'prop-types';
import get from 'lodash/get';
import classnames from 'classnames';
import Text from 'atoms/Text';
import ProgressBarStep from './ProgressBarStep';
import styles from './step_indicator.module.scss';

function StepIndicator(props) {
  const {
    className,
    steps,
    navigateToPath
  } = props;

  const activeStep = steps.find(step => step.currentStepActive) || steps[0];

  return (
    <div className={classnames(className)}>
      <Text tag='div' type={10} color='neutral-2' font='a' className={styles['active-step']}>
        {get(activeStep, 'text', '')}
      </Text>
      <div className={styles['step-indicator']}>
        {steps.map((step, idx) =>
          <ProgressBarStep
            key={idx}
            step={step}
            navigateToPath={navigateToPath}
          />
        )}
      </div>
    </div>
  );
}

StepIndicator.propTypes = {
  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,
  /**
   * An array of the steps to display
   */
  steps: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      route: PropTypes.string,
      nextStepAccessible: PropTypes.bool,
      currentStepActive: PropTypes.bool,
      clickable: PropTypes.bool,
    })
  ).isRequired,
  /**
   * Click handler for steps; expects to be called with path name
   */
  navigateToPath: PropTypes.func.isRequired
};

export default StepIndicator;
