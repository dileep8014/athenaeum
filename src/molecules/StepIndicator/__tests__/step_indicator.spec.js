import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import StepIndicator from 'molecules/StepIndicator';

describe('<StepIndicator />', () => {
  it('renders correctly', () => {
    const wrapper = mount(<StepIndicator steps={[]} navigateToPath={() => {}} />);

    expect(wrapper.type()).toEqual(StepIndicator);
  });

  it('renders correctly', () => {
    const actual = renderer.create(<StepIndicator steps={[]} navigateToPath={() => {}} />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
