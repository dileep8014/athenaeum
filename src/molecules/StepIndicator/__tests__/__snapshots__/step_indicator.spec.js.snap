// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepIndicator /> renders correctly 1`] = `
<div
  className="step-indicator"
/>
`;
